namenode/src/main/java/org/apache/crail/namenode/BlockStore.java

```diff
@@ -222,6 +303,33 @@ short removeDatanode(DataNodeInfo dn) throws IOException {
 
 	//---------------
 
+	public long getTotalCapacity() {
+		long capacity = 0;
+
+		for(DataNodeBlocks datanode : membership.values()) {
+			capacity += datanode.getMaxBlockCount();
+		}
+
+		return capacity;
+	}
+
+	public long getFreeCapacity() {
+		long capacity = 0;
+
+		for(DataNodeBlocks datanode : membership.values()) {
+			capacity += datanode.getBlockCount();
+		}
+
+		return capacity;
+	}
+
+	public Collection<DataNodeBlocks> getDataNodeBlocks() {
+		return this.membership.values();
+	}
+
+	public int getRunningDatanodes() {
```

Comment:
Naming. Maybe `getNumberOfRunningDatanodes()`?
I know its long but it could be misinterpreted otherwise.